shells/chrome/src/devtools-background.js

```diff
@@ -24,10 +27,65 @@ function createPanelIfHasVue () {
       created = true
       chrome.devtools.panels.create(
         'Vue', 'icons/128.png', 'devtools.html',
-        function (panel) {
+        panel => {
           // panel loaded
+          panel.onShown.addListener(onPanelShown)
+          panel.onHidden.addListener(onPanelHidden)
         }
       )
     }
   )
 }
+
+// Manage panel visibility
+
+function onPanelShown () {
+  chrome.runtime.sendMessage('vue-panel-shown')
+  panelShown = true
+}
+
+function onPanelHidden () {
+  chrome.runtime.sendMessage('vue-panel-hidden')
+  panelShown = false
+}
+
+// Page context menu entry
+
+chrome.contextMenus.create({
+  id: 'vue-inspect-instance',
+  title: 'Inspect Vue component',
+  contexts: ['all']
+})
+
+chrome.contextMenus.onClicked.addListener(genericOnContext)
+
+function genericOnContext (info, tab) {
+  if (info.menuItemId === 'vue-inspect-instance') {
+    panelAction(() => {
+      chrome.runtime.sendMessage('vue-get-context-menu-target')
+    })
+  }
+}
+
+// Action that may execute immediatly
+// or later when the Vue panel is ready
+
+function panelAction (cb) {
+  if (created && panelShown) {
+    cb()
+  } else {
+    pendingAction = cb
+  }
+}
+
+// Execute pending action when Vue panel is ready
+
+chrome.runtime.onMessage.addListener(request => {
+  if (request === 'vue-panel-load') {
+    onPanelLoad()
+  }
+})
+
+function onPanelLoad () {
+  pendingAction && pendingAction()
```

Comment:
when does the pendingAction gets executed? I couldn't test this one
Depending on when it is triggered, it may be better to set `pendingAction` back to `null`

Reply:
Fixed